litgpt/api.py

```diff
@@ -185,30 +188,53 @@ def generate(
                 "Support for multiple devices is currently not implemented for `generate`"
             )
 
-        output_ids = generate_fn(
-            model=self.model,
-            prompt=input_ids.to(self.fabric.device),
-            max_returned_tokens=max_returned_tokens,
-            temperature=temperature,
-            top_k=top_k,
-            top_p=top_p,
-            eos_id=self.preprocessor.tokenizer.eos_id,
-            include_prompt=include_prompt,
-        )
+        def iterator():
+            outputs = stream_generate(
+                model=self.model,
+                prompt=input_ids.to(self.fabric.device),
+                max_returned_tokens=max_returned_tokens,
+                temperature=temperature,
+                top_k=top_k,
+                top_p=top_p,
+                stop_tokens=([self.preprocessor.tokenizer.eos_id],),
+            )
+            if return_as_token_ids:
+                yield from outputs
+            else:
+                for output in outputs:
+                    yield self.preprocessor.tokenizer.decode(output)
+            return
+
+        if stream:
+            outputs = iterator()
+        else:
+            outputs = generate_fn(
+                model=self.model,
+                prompt=input_ids.to(self.fabric.device),
+                max_returned_tokens=max_returned_tokens,
+                temperature=temperature,
+                top_k=top_k,
+                top_p=top_p,
+                eos_id=self.preprocessor.tokenizer.eos_id,
+                include_prompt=include_prompt,
+            )
 
         for block in self.model.transformer.h:
             block.attn.kv_cache.reset_parameters()
 
+        if stream:
+            return outputs
         if return_as_token_ids:
```

Comment:
Not necessary, but I think 

```suggestion
        elif return_as_token_ids:
```

makes it perhaps a bit more readable. In case there gets more code added over time between this and the `iterator()` function above.